File: 2020/day1/answer.py
from itertools import combinations

with open('input') as f:
    input = [int(x) for x in f.read().split('\n')]

    # Part 1
    answer1 = [x[0] * x[1] for x in list(combinations(input, 2)) if sum(x) == 2020 ][0]

    # Part 2
    answer2 = [x[0] * x[1] * x[2] for x in list(combinations(input, 3)) if sum(x) == 2020 ][0]

    print(f'Part 1: {answer1}')
    print(f'Part 2: {answer2}')

File: 2020/javascript/day1/answer.js
const fs = require('fs')

const input = fs.readFileSync('input', { encoding: 'utf8', flag: 'r' }).split('\n').map(x => parseInt(x))

function getProductPair(input, sum) {
  const found = new Set()
  for (const i of input) {
    let pair = sum - i
    
    if (found.has(pair)) {
      return i * pair
    }
    found.add(i)
  }
}

function getProductTriplet(input, sum) {
  for (let i = 0; i < input.length - 2; i++) {
    const pairSum = sum - input[i]
    const found = new Set()
    for (let j = i + 1; j < input.length; j++) {
      let pair = pairSum - input[j]

      if (found.has(pair)) {
        return input[i] * input[j] * pair
      }
      found.add(input[j])
    }
  }
}

console.log(`Part 1: ${getProductPair(input, 2020)}`)
console.log(`Part 2: ${getProductTriplet(input, 2020)}`)